### DISCUSSION TITLE
How to use touchSensors?

### DISCUSSION THREAD

**Discussion by GramGra07** [FIRST-Tech-Challenge/FtcRobotController#504] [Q&A] [answered]
I have been having problems with coding my REV touch sensors, before updating everything to 8.1 from 8.0, I had my touch sensors coded as ```public TouchSensor = touchSensor;``` and ```touchSensor = hardwareMap.get(TouchSensor.class,"touchSensor");``` and this was working. I updated, and when I looked at the configuration on the driver hub, it had lost the configuration I had setup (Control Hub - Digital - 1 - TouchSensor "touchSensor"). I tried the example of the digital touch concept, and it gave me an error saying it couldn't use the RunMode Input for digitalChannel on a null object reference. For reference ```DigitalChannel touchSensor;``` ```touchSensor = hardwareMap.get(DigitalChannel.class, "touchSensor");``` ```touchSensor.setMode(DigitalChannel.Mode.INPUT);``` ```if (touchSensor.getState() == true)...```. I just saw that there is an issue concerning this but in the meantime i'm not sure how to fix this or what to do differently. Any tips?

**Comment by cmacfarl** [comment, marked as answer]
If you are not using the IMU on a Control Hub purchased after Sept 2022, then I would recommend you simply revert to 8.0.  There is nothing in 8.1 that is absolutely mandatory.  8.1 is an optional release that primarily provides support for the new IMU chip in the Control Hubs released this fall.

8.1.1 with a fix for this is in progress, but we don't an exact date yet.

If you are using the REV Hardware Client to update, there is a dropdown that you can use to "update" to a previous version.

If you are using GitHub to download and side load APKs via adb you can just download and side load the v8.0 APK.  https://github.com/FIRST-Tech-Challenge/FtcRobotController/releases/tag/v8.0

If you have cloned the repository, and have a local git instance, you can checkout the v8.0 tag, or you can roll back the merge for v8.1.  How you proceed in this instance depends upon how you manage your local fork/clone.

**Reply by GramGra07** [reply]
Ok thanks!!
